docs/DeveloperGuide.md

```diff
@@ -104,38 +109,42 @@ This section describes some details on how some features are implemented.
 
 ### Tag Feature
 
-The tag command accepts a string from the user and adds it into `ArrayList<String> tags` of a `Task`.  
-
-
-Here is an example on adding a tag to a general task:  
-
-1. User inputs `tag add 2 "testTag"`. 
-2. `TagParser` will initialise `TagCommand` with add as `tagOperation` 2 as `taskIndex` and testTag as `tagDescription`, while `taskModule` is null.
-3. `TagCommand` then gets the relevant `Module`. If `taskModule` is null, `getGeneralTasks()` is called. Else, `getModule(taskModule)` is called instead.
-4. Next, `TagCommand` checks the `tagOperation`. If add, `addTag(targetModule)` is called. Else if del, `removeTag(targetModule)` is called. Else, it throws `ParseException`.
+The tag command accepts a string from the user and adds it into the `tags` attribute (an `ArrayList<String>`) of the specified `Task`.
 
-Below is the sequence diagram of how the tag feature works:
+The following sequence diagram illustrates the process:
 
 ![Sequence Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/TagSeqDiagram/Tag.puml)
+
 ![Sequence Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/TagSeqDiagram/GetModule.puml)
+
 ![Sequence Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/TagSeqDiagram/CheckAndRunTagOperation.puml)
 
+Here is an example on adding a tag to a general task:  
+
+1. User inputs `tag add 2 "testTag"`. 
+2. `ModHappyParser` identifies the command word as `tag` and passes `add 2 "testTag"` to `TagParser`.
+3. `TagParser` instantiates a `TagCommand` with `tagOperation = "add"`, `taskIndex = 2`, `tagDescription = "testTag"` and `taskModule = null`. This is returned to `Main`.
+4. `Main` calls the `execute()` method of the `TagCommand` instance.
+5. `TagCommand` first gets the relevant `Module`. Since `taskModule` is null, `getGeneralTasks()` is called and the General Tasks `Module` is retrieved.
+6. Next, `TagCommand` checks the `tagOperation`. As its value is `add`, `addTag(targetModule)` is called.
+7. Finally, command feedback is returned to `Main`, indicating that the operation was successful.
+
 ### GPA Feature
 
-The gpa command takes in a single command word `gpa` and no other arguments from the user (unlike the tag command above), and returns user's GPA in 2 decimal places.
+The `gpa` command takes in a single command word `gpa` and no other arguments from the user, returning the user's GPA to 2 decimal places.
```

Comment:
Should this be GpaCommand instead if it's surrounded by ``

Reply:
Comparing with the Tag Feature section above, there is no mention about what the command word / format is, only what it does. So I will probably change it to match. Something like "this feature calculates the user's GPA to 2 decimal places, based on the modules stored in the module list"

Reply:
okk, cuz I felt it more meaningful to surround it with `` only if the "word" is some name exactly in our code or in any specific rule.